### PR TITLE
Ensure correct NDK version is used for test fixture modules

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Ensure all e2e test fixture modules use the correct version of the NDK.

## Changeset

- Gradle file updates
- CI pipeline and Dockerfile updates to set/provide NDK versions appropriately.

## Testing

Covered by CI - in particular, I have verified that the output for each test fixture build step indicates that the NDK version has been set correctly.

### CODE REVIEW

File: features/fixtures/mazerunner/build.gradle [hunk 1]
```diff
@@ -28,3 +28,12 @@ buildscript {
...
     }
 }
+
+if (project.hasProperty("TEST_FIXTURE_NDK_VERSION")) {
+    def version = project.property("TEST_FIXTURE_NDK_VERSION")
+    project.logger.lifecycle("Building test fixture with custom NDK version '$version'")
+    ext.ndkVersion = version
+} else {
+    project.logger.lifecycle("Building test fixture with default NDK version")
+    ext.ndkVersion = "17.2.4988734"
```
Comment (kattrali) [inline features/fixtures/mazerunner/build.gradle:38]: Any idea why r17 is used here? @lemnik?

**Review by kattrali — APPROVED** [review]

**Review by lemnik — APPROVED** [review]
LGTM

File: features/fixtures/mazerunner/build.gradle (hunk 1 above)
Comment (lemnik) [reply features/fixtures/mazerunner/build.gradle:38]: It happened as part of a move to AGP 7, although I'm not entirely clear why.

**Review by lemnik — APPROVED** [review]
LGTM

**Pull request merged**